### ISSUE TITLE
Error in Wireshark make 

### ISSUE THREAD

**Issue by laithmatalkah** [AFITWiSec/EZ-Wave#3]
when executing make command in Wiresark the following errors have been occurred, it seems that these errors came from the copied files from setup/wireshark to   /epan/dissectors
"packet-afit-encapse.c: In function 'proto_register_afit_encap':
packet-afit-encapse.c:141:155: error: 'DISSECTOR_TABLE_NOT_ALLOW_DUPLICATE' undeclared (first use in this function)
    afit_encap_dissector_table = register_dissector_table("afit_encap.encap_type", "Temporary Encapsulation Type for ZWAVE dissector", FT_UINT8, BASE_DEC, DISSECTOR_TABLE_NOT_ALLOW_DUPLICATE); 
                                                                                                                                                           ^
packet-afit-encapse.c:141:155: note: each undeclared identifier is reported only once for each function it appears in
packet-afit-encapse.c:141:4: error: too many arguments to function 'register_dissector_table'
    afit_encap_dissector_table = register_dissector_table("afit_encap.encap_type", "Temporary Encapsulation Type for ZWAVE dissector", FT_UINT8, BASE_DEC, DISSECTOR_TABLE_NOT_ALLOW_DUPLICATE); 
    ^
In file included from packet-afit-encapse.c:21:0:
../../epan/packet.h:206:33: note: declared here
 WS_DLL_PUBLIC dissector_table_t register_dissector_table(const char *name,
                                 ^
packet-afit-encapse.c: In function 'proto_reg_handoff_afit_encap':
packet-afit-encapse.c:152:5: warning: passing argument 1 of 'create_dissector_handle' from incompatible pointer type [enabled by default]
     afit_encap_handle = create_dissector_handle (dissect_afit_encap, proto_afit_encap);
     ^
In file included from packet-afit-encapse.c:21:0:
../../epan/packet.h:507:34: note: expected 'dissector_t' but argument is of type 'int (*)(struct tvbuff_t *, struct packet_info *, struct proto_tree *, void *)'
 WS_DLL_PUBLIC dissector_handle_t create_dissector_handle(dissector_t dissector,
                                  ^
make[5]: *** [libdissectors_la-packet-afit-encapse.lo] Error 1
make[5]: Leaving directory `/home/me/wireshark-2.0.2/epan/dissectors'
make[4]: *** [all-recursive] Error 1
make[4]: Leaving directory `/home/me/wireshark-2.0.2/epan/dissectors'
make[3]: *** [all] Error 2
make[3]: Leaving directory `/home/me/wireshark-2.0.2/epan/dissectors'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/home/me/wireshark-2.0.2/epan'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/me/wireshark-2.0.2'
make: *** [all] Error 2"

**Comment by eficker** [comment]
Same exact error here.

**Comment by freb** [comment]
Me too. I really feel like we're all in this together given that we all hit this same issue today today :)

**Comment by gitjhall** [comment]
What versions of Wireshark are you guys installing? I think it problem comes from an update to Wireshark (something similar happened to me).

We've been adding to the dissector to add Z-Wave routing. When we push that we'll make sure this bug is addressed.

**Comment by eficker** [comment]
I was compiling the lastest wireshark versions from both their repo and the Debian repo. 

Is there a release version that is known working with the current EZ-Wave code?

**Comment by eficker** [comment]
I just confirmed that this also fails with wireshark releases 1.12.9 and 2.0.0.

**Comment by laithmatalkah** [comment]
I used both 1.12 and 2.0 stable and old stable release
I've download the Zip file 
Extract
then follow installation steps.
This error  doesn't show up when I ignore the ./autogen.sh command

**Comment by eficker** [comment]
@gitjhall Is there a version of wireshark that is known working? I've got a little time sensitive research I need to wrap up.

**Comment by gitjhall** [comment]
I don't have that box with me, but it was built from source using v2.0.1.x
You also don't need the wireshark dissector to sniff or inject traffic. If you're successfully capturing z-wave traffic with scapy-radio, you can use packet.show() in whatever Python script you're running to get a printout of the z-wave packet.

**Comment by sumeetjauhar** [comment]
I have the same error as all of you guys. Let me give v2.0.1 a try . Let me know if you guys got it working with another version.

I shall update you guys soon.

**Comment by sumeetjauhar** [comment]
Seems like v2.0.1 does not work either . Anyone know how to avoid this error ?

In file included from packet-afit-encapse.c:21:0:
../../epan/packet.h:507:34: note: expected 'dissector_t' but argument is of type 'int (*)(struct tvbuff_t *, struct packet_info *, struct proto_tree *, void *)'
 WS_DLL_PUBLIC dissector_handle_t create_dissector_handle(dissector_t dissector,
                                  ^
make[5]: *** [libdissectors_la-packet-afit-encapse.lo] Error 1
make[5]: Leaving directory `/home/osboxes/wireshark-2.0.1/epan/dissectors'
make[4]: *** [all-recursive] Error 1

**Comment by laithmatalkah** [comment]
"I don't have that box with me, but it was built from source using v2.0.1.x
You also don't need the wireshark dissector to sniff or inject traffic. If you're successfully capturing z-wave traffic with scapy-radio, you can use packet.show() in whatever Python script you're running to get a printout of the z-wave packet."

can you explain more details???

**Comment by eficker** [comment]
I really just needed access to raw packet data, the debug output in the GRC
file gave me that.

Awesome work, keep it up!

**Comment by cwbadenhop** [comment]
The packet-afit-encapse.c file in the github repo is a forked version I used for an exam and is the wrong sized header for the vanilla Zwave.grc and associated grc C files. While it does not explain the compile errors, I will push a version of the dissector files that compiles on my machine with wireshark 2.0.1 src as soon as I can permission from @gitjhall. In the mean time, here is a tar.gz of the update
[wireshark.tar.gz](https://github.com/AFITWiSec/EZ-Wave/files/192327/wireshark.tar.gz)

**Comment by laithmatalkah** [comment]
Thanks cwbadenhop it  works without errors . Now having solved this problem , can any body tell me where should I start to be able to sniff Zwave packets.

**Comment by gitjhall** [comment]
@laithmatalkah please start a new issue requesting help sniffing traffic so we can keep that topic separate from compiling the Wireshark dissector.
